rs/nns/governance/tests/governance.rs

```diff
@@ -1805,6 +1811,300 @@ async fn test_manage_network_economics_change_one_deep_subfield() {
+                    neurons_fund_economics: Some(NeuronsFundEconomics {
+                        maximum_icp_xdr_rate: Some(Percentage {
+                            basis_points: Some(new_max),
+                        }),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                })),
+            },
+        )
+        .await
+        .unwrap();
+
+    // Step 2.2: Vote in the two proposals. This unleashes the chaos: the first
+    // one works, but that causes the second one to fail.
+    for proposal_id in [set_min_proposal_id, set_max_proposal_id] {
+        let result = gov
+            .manage_neuron(
+                &controller,
+                &ManageNeuron {
+                    command: Some(manage_neuron::Command::RegisterVote(
+                        manage_neuron::RegisterVote {
+                            proposal: Some(proposal_id),
+                            vote: Vote::Yes as i32,
+                        },
+                    )),
+                    neuron_id_or_subaccount: Some(manage_neuron::NeuronIdOrSubaccount::NeuronId(
+                        NeuronId { id: 2 },
+                    )),
+                    id: None,
+                },
+            )
+            .await;
+        // Assert vote successful.
+        match result.command {
+            Some(manage_neuron_response::Command::RegisterVote(_)) => (),
+            _ => panic!("{:#?}", result),
+        }
+    }
+
+    // Step 3: Verify results.
+
+    // Step 3.1: Only the frist proposal executed successfully.
+
+    let set_min_proposal = gov
+        .get_proposal_info(&controller, set_min_proposal_id)
+        .unwrap();
+    assert_ne!(
+        set_min_proposal.decided_timestamp_seconds, 0,
+        "{:#?}",
+        set_min_proposal,
+    );
+    assert_ne!(
+        set_min_proposal.executed_timestamp_seconds, 0,
+        "{:#?}",
+        set_min_proposal,
+    );
+    assert_eq!(
+        set_min_proposal.failure_reason, None,
+        "{:#?}",
+        set_min_proposal,
+    );
+
+    let set_max_proposal = gov
+        .get_proposal_info(&controller, set_max_proposal_id)
+        .unwrap();
+    // Yes, it was decided, like the other proposal.
+    assert_ne!(
+        set_max_proposal.decided_timestamp_seconds, 0,
+        "{:#?}",
+        set_max_proposal,
+    );
+    // Sadly, though, it failed, unlike the other proposal.
+    assert_ne!(
+        set_max_proposal.failed_timestamp_seconds, 0,
+        "{:#?}",
+        set_max_proposal,
+    );
+    // And the reason it failed was...
+    let failure_reason = set_max_proposal.failure_reason.clone().unwrap();
+    assert_eq!(
+        ErrorType::try_from(failure_reason.error_type),
+        Ok(ErrorType::InvalidProposal),
+        "{:#?}",
+        set_max_proposal,
+    );
+    let message = failure_reason.error_message.to_lowercase();
+    for key_word in [
```

Comment:
NCR; We keep implementing fuzzy matching for out tests, I wonder if instead we should use a library for that.